### ISSUE TITLE
[Module Proposal]: `avm-res-sql-instancepool` (tf)

### ISSUE THREAD

**Issue by PmeshramPM** [Azure/Azure-Verified-Modules#531]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Terraform

### Module Classification?

Resource Module

### Module Name

avm-res-sql-instancepool

### Module Details

AVM Resource Module for SQL Instance Pools

### Do you want to be the owner of this module?

No

**Comment by sai-lalith** [comment]
Hey, I would like to be this module's owner.

**Comment by yashints** [comment]
I am happy to be the owner of this @PmeshramPM

**Comment by PmeshramPM** [comment]
Hi @yashints ,

Thanks for requesting/proposing to be an AVM module owner!

We just want to confirm **you agree to the below pages** that define what module ownership means:

- [Team Definitions & RACI](https://azure.github.io/Azure-Verified-Modules/specs/shared/team-definitions)
- [Shared Specification (Bicep & Terraform)](https://azure.github.io/Azure-Verified-Modules/specs/shared)
- [Module Support](https://azure.github.io/Azure-Verified-Modules/help-support/module-support)

Any questions or clarifications needed, let us know!

If you agree, please just **reply to this issue with the exact sentence below** (as this helps with our automation 👍):

"I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER"

Thanks,

The AVM Core Team

#RR

**Comment by yashints** [comment]
@PmeshramPM do we have a repo for the Bicep version? Would I be the owner of that too?

**Comment by PmeshramPM** [comment]
> @PmeshramPM do we have a repo for the Bicep version? Would I be the owner of that too?

Thanks @yashints, This Issue only captures the Terraform development. If you are looking to develop a Bicep equivalent, please raise another issue with Bicep naming conventions.

**Comment by yashints** [comment]
@PmeshramPM seems like there is no terraform provider for the instance pool, what's the approach for this situation?

**Comment by PmeshramPM** [comment]
@yashints - We have checked internally and you could use AzAPI provider - https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/instancepools?pivots=deployment-language-terraform

**Comment by yashints** [comment]
@PmeshramPM I've raised a PR, however, there seems to be an issue with the script which checks the linting for terraform which is also discussed on the AVM terraform teams channel

**Comment by yashints** [comment]
@PmeshramPM this is blocked as I cannot proceed without the core team letting me know how to fix the e2e tests when a resource takes more than an hour to get created and also destroying the resources won't happen immediately

**Comment by PmeshramPM** [comment]
> e AVM terraform Teams channel

@yashints - I am following up on this one and will confirm

**Comment by yashints** [comment]
this is still blocked...

**Comment by yashints** [comment]
This is still in blocked state, waiting for the AVM team to come back with what's next

**Comment by prjelesi** [comment]
@yashints lets sync offline to see how we can unblock this.

**Comment by matebarabas** [comment]
@yashints, as the @Azure/avm-core-team-technical-terraform team stated in our internal discussions: 

_After consideration we have decided that this resource is not suitable for use as a terraform module due to the deletion issues.
We therefore have decided this is blocked until either:_
- _the RP rectifies the soft deletion issue_
- _the resource is implemented in AzureRM where the delete can be tracked and polled_

I'm marking this issue as long term and will disassociate it from the current owner.

**Comment by sujaypillai** [comment]
I CONFIRM I WISH TO OWN THIS AVM MODULE AND UNDERSTAND THE REQUIREMENTS AND DEFINITION OF A MODULE OWNER.

**Comment by jeanchg** [comment]
Hi @sujaypillai ,

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉 If you have additional contributors, ensure you grant them permissions by adding them to the related GitHub teams, as outlined [here](https://azure.github.io/Azure-Verified-Modules/spec/SNFR20).

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by jeanchg** [comment]
@sujaypillai the repo isn't created yet, pls follow the guidance https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/repository-creation-process/ to do so. Let me know if you face any issue.

**Comment by sujaypillai** [comment]
@jeanchg - The module repository has now been created. You can find it at [https://github.com/Azure/terraform-azapi-avm-res-sql-instancepool](https://github.com/Azure/terraform-azapi-avm-res-sql-instancepool)
